integration_tests/src/helpers/exchangeRate.ts

```diff
@@ -0,0 +1,153 @@
+  const tokenContractConfig = await clientCW.queryContractSmart(
+    coreConfig.token_contract,
+    {
+      config: {},
+    },
+  );
+  const { amount } = await queryService.SupplyOf({
+    denom: tokenContractConfig.denom,
+  });
+  let exchangeRateDenominator: Decimal = new Decimal(amount.amount);
+  if (exchangeRateDenominator.isZero()) {
+    return new Decimal(1).toNumber();
+  }
+  const delegationsResponse = await clientCW.queryContractSmart(
+    coreConfig.puppeteer_contract,
+    {
+      extension: {
+        msg: {
+          delegations: {},
+        },
+      },
+    },
+  );
+  const delegationsAmount: Decimal =
+    delegationsResponse.delegations.delegations.reduce(
+      (acc: Decimal, next: any) => acc.plus(new Decimal(next.amount.amount)),
+      new Decimal(0),
+    );
+  const batchID = await clientCW.queryContractSmart(coreContract, {
+    current_unbond_batch: {},
+  });
+  const batch = await clientCW.queryContractSmart(coreContract, {
+    unbond_batch: {
+      batch_id: String(batchID),
+    },
+  });
+  let unprocessedDassetToUnbond: Decimal = new Decimal('0');
+  if (batch.status === 'new') {
+    unprocessedDassetToUnbond = unprocessedDassetToUnbond.plus(
+      batch.total_dasset_amount_to_withdraw,
+    );
+  }
+  if (Number(batchID) > 0) {
+    const penultimate = Number(batchID) - 1;
+    const penultimateBatch = await clientCW.queryContractSmart(coreContract, {
+      unbond_batch: {
+        batch_id: String(penultimate),
+      },
+    });
+    if (penultimateBatch.status === 'unbond_requested') {
+      unprocessedDassetToUnbond = unprocessedDassetToUnbond.plus(
+        batch.total_dasset_amount_to_withdraw,
+      );
+    }
+  }
+  const failedBatchID = await clientCW.queryContractSmart(coreContract, {
+    failed_batch: {},
+  });
+  if (failedBatchID.response !== null) {
+    const failedBatch = await clientCW.queryContractSmart(coreContract, {
+      batch_id: failedBatchID.response,
+    });
+    unprocessedDassetToUnbond = unprocessedDassetToUnbond.plus(
+      failedBatch.total_dasset_amount_to_withdraw,
+    );
+  }
+  exchangeRateDenominator = exchangeRateDenominator.plus(
+    unprocessedDassetToUnbond,
+  );
+  const stakerBalance: Decimal = new Decimal(
+    await clientCW.queryContractSmart(coreConfig.staker_contract, {
+      all_balance: {},
+    }),
+  );
+  const totalLSMShares: Decimal = new Decimal(
+    await clientCW.queryContractSmart(coreContract, {
+      total_l_s_m_shares: {},
+    }),
+  );
+  const exchangeRateNumerator: Decimal = delegationsAmount
+    .plus(stakerBalance)
+    .plus(totalLSMShares);
+  if (exchangeRateNumerator.isZero()) {
+    return 1;
+  }
+  const exchangeRate: Decimal = exchangeRateNumerator.dividedBy(
+    exchangeRateDenominator,
+  );
+  return exchangeRate.toNumber();
+}
+
+export async function checkExchangeRate(testContext: any) {
+  const { coreContractClient, client, neutronRPCEndpoint } = testContext;
```

Comment:
please use coreContractClient, puppeterContractClient, etc to query data. no need to connect once again